content/blog/jamstack-with-pulumi/index.md

```diff
@@ -0,0 +1,359 @@
+
+```bash
+$ wget https://github.com/budparr/gohugo-theme-ananke/archive/master.zip
+$ unzip master.zip
+$ cp -rf gohugo-theme-ananke-master ./themes/ananke
+$ rm -rf master.zip gohugo-theme-ananke-master/
+```
+
+Configure the site by editing the *config.toml* file with the following.
+
+```toml
+baseURL = "http://www.<your_domain>.com/"
+languageCode = "en-us"
+title = "Platypus"
+theme = "ananke"
+```
+
+Use hugo to create a new markdown page in the posts directory.
+
+```bash
+$ hugo new posts/platypus.md
+```
+
+Now it’s time to add content to your site. You can use your favorite editor to write and edit content in your working directory on your machine instead of using the container shell. Edit the markdown file, below is an example page but feel free to add your content.
+
+```md
+## Platypus of the Day
+
+**This platypus is a cool critter!**
+
+![platypus](https://static01.nyt.com/images/2017/08/01/science/29TB-PLATYPUS1/29TB-PLATYPUS1-superJumbo.jpg)
+```
+
+Finally, we can build the static site with Hugo in the container command line.
+
+```bash
+$ hugo -D
+```
+
+Hugo will create a *public* directory with the static website.
+
+## Building the Cloud Infrastructure
+
+Configuring a static website can be complicated when using either the AWS web interfaces or the AWS CLI. To deploy a static website, we need to create and configure the following cloud resources.
+
+- [Amazon S3](https://aws.amazon.com/s3/) to store the website's contents
+- [Amazon CloudFront](https://aws.amazon.com/cloudfront/) for the CDN serving content
+- [Amazon Route53](https://aws.amazon.com/route53/) to set up the DNS for the website
+- [Amazon Certificate Manager](https://aws.amazon.com/certificate-manager/) for securing the site via HTTPS
+
+Using Pulumi to deploy Infrastructure as Code, we can create and build our infrastructure using Python or your favorite programming language. If you haven't installed Pulumi and configured it to work with your AWS credentials, follow the [Getting Started with AWS guide]({{< relref "/docs/get-started/aws">}}).
+
+To get started building our infrastructure, we’ll download the Python example for setting up a static website with AWS. There are many [examples on our GitHub repository](https://github.com/pulumi/), but we can clone just the [AWS Python Static Website example](https://github.com/pulumi/examples/tree/master/aws-py-static-website) using a sparse clone:
+
+```bash
+$ mkdir aws-website
+$ cd aws-website
+
+$ git init
+$ git remote add origin -f https://github.com/pulumi/examples/
+$ git config core.sparseCheckout true
+$ echo aws-py-static-website >> .git/info/sparse-checkout
+$ git pull origin master
+```
+
+## Configuring and Deploying the Website
+
+Now that we have a project to build the website, we will need to configure it so that it uses your domain name. Here are the steps to get it running.
+
+1. Create a new stack or instance for testing:
+
+    ```bash
+    $ pulumi stack init website-testing
```

Comment:
When I run this command in the directory I was in when I left off from the last step above, I get an error about there not being a Pulumi project. We should probably add a `cd aws-py-static-website`.